packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -0,0 +1,124 @@
+  // const prompt1 = `Today's date is ${todaysDate}. Write a short summary of the patient's well-being that is relevant today. Be specific with the dates for any significant events. Focus on any diagnoses that occurred in the past year.`;
+  // const prompt2 = `Today's date is ${todaysDate}. Given the attached medical documents, summarize them into a concise summary of the patients medical history. In your first sentence, include the patient's age/gender and all significant medical conditions. In the second sentence, summarize any notable recent health events (eg hospitalizations or emergency department visits) and any requested follow-up. Do not include directions on taking medications. Here is an example of an ideal summary: Patient is a 65 yo male with hx of HTN, poorly controlled diabetes, and smoking who presented with 3 hours of crushing substernal chest pain. They underwent a cardiac cath with stent placement and were discharged on aspirin and metoprolol.`;
+  const body = processedBundleInput;
+  try {
+    const brief = bedrockUtils.getBedrockResponse({
+      prompt,
+      body,
+    });
+    log(`Brief generated.`);
+    return brief;
+  } catch (error) {
+    log(`Error generating brief: ${errorToString(error)}`);
+    return undefined;
+  }
+}
+
+export async function prepareBundleForBrief(bundle: Bundle): Promise<string | undefined> {
+  const bundleCopy = { ...bundle };
+
+  if (!bundle.entry?.length) return undefined;
+  const dedupedBundle = deduplicateBundleResources(bundleCopy);
+  if (!dedupedBundle.entry?.length) return undefined;
+  const filteredString = filterBundleResources(dedupedBundle);
+  if (!filteredString) return undefined;
+  return (await preprocessDataWithRag(filteredString)).trim();
+}
+
+function deduplicateBundleResources(bundle: Bundle): Bundle {
+  // TODO: Implement FHIR deduplication algorithm
+  return bundle;
+}
+
+function filterBundleResources(bundle: Bundle): string | undefined {
+  // TODO: Implement a filter for only including inputs within a certain time frame
+  const entries = bundle.entry;
+  const filteredEntries = entries?.filter(entry => {
+    const resource = entry.resource;
+    if (resource && relevantResources.includes(resource.resourceType)) {
+      return true;
+    }
+    return false;
+  });
+
+  if (!filteredEntries) return undefined;
+  bundle.entry = filteredEntries;
+
+  const patient = findPatientResource(bundle);
+  if (!patient) return undefined;
+  const ptData = `PT family name: ${patient.name?.[0]?.family}, given: ${patient.name?.[0]?.given}, DOB: ${patient.birthDate}`;
```

Comment:
This is going straight to the model, right? Any particular reason to use PT and DOB?

Reply:
Yes. I played around with sending some data with and without it, and this seems to help the model get the general information right, compared to when this isn't included. I changed `PT` and `DOB` to `Patient` and `date of birth`, respectively.